RCode/NEFI/partition_uncertainty_DayLengthQuad.R

```diff
@@ -0,0 +1,274 @@
+N.cols <- c("black","red","green","blue","orange") ## set colors
+ylim = range(ci+0.01, na.rm = TRUE)
+trans <- 0.8       ## set transparancy
+cal_time = as.Date(as.character(dat$cal$date))
+forecast_time = as.Date(as.character(dat$forecast$date))
+all_time = c(cal_time, forecast_time)
+
+plot.run <- function(){
+  plot(all_time, all_time, type='n', ylim=ylim, ylab="Gloeo count", log = 'y')
+  axis(2)
+  ecoforecastR::ciEnvelope(cal_time, ci[1,], ci[3,], col="lightBlue")
+  lines(cal_time, ci[2,], col="purple")
+  points(cal_time, ci[2,])
+}
+
+#ci <- apply(exp(out[ ,mus]), 2, quantile, c(0.025,0.5,0.975))
+plot.run()
+
+
+#setting up deterministic forecast function for random walk 
+# IC = initial conditions from end of calibration period 
+# N_out = number of time steps to forecast 
+# tau_add = process error (defaults to zero) 
+# n = number of monte carlo sims 
+forecast_gloeo <- function(IC, beta1, beta2, beta3, beta4, DL=DL, N_out = N_out, tau_add = 0, n = Nmc){
+  out <- matrix(NA, n, N_out) # setting up output 
+  gloeo_prev <- IC
+  for(i in 1:N_out){
+    lambda <- beta1 + beta2*gloeo_prev + beta3*DL[i] + beta4*DL[i]^2
+    out[,i] <- rnorm(n,lambda,tau_add) 
+    gloeo_prev <- out[,i] # update IC 
+  }
+  return(out)
+}
+
+## deterministic predictions 
+DL = c(dat$forecast$daylength) #DL during forecasted period
+N_out = nrow(dat$forecast) # length of forecast time points 
+IC = out[,mus]
+betas = out[,grep("beta",colnames(out))]
+betas.mean <- apply(betas,2,mean)
+
+gloeo.det <- forecast_gloeo(IC = mean(IC[,N]), # last column of cal is the initial condition for forecast 
+                            N_out = N_out,
+                            beta1 = betas.mean[1],
+                            beta2 = betas.mean[2],
+                            beta3 = betas.mean[3],
+                            beta4 = betas.mean[4],
+                            DL = DL,
+                            tau_add = 0,
+                            n = 1)
+
+## Plot run
+plot.run()
+lines(forecast_time, exp(gloeo.det), col="purple", lwd=3)
+
+
+######## initial condition uncertainty #######
+# sample rows from previous analysis
+prow = sample.int(nrow(out),Nmc,replace=TRUE)
+
+gloeo.I <- forecast_gloeo(IC = IC[prow, N],
+                      N_out = N_out,
+                      tau_add = 0,
+                      beta1 = betas.mean[1],
+                      beta2 = betas.mean[2],
+                      beta3 = betas.mean[3],
+                      beta4 = betas.mean[4],
+                      DL = DL,
+                      n = Nmc)
+gloeo.I.ci = apply(exp(gloeo.I), 2, quantile, c(0.025,0.5,0.975))
+
+## Plot run
+plot.run()
+ecoforecastR::ciEnvelope(forecast_time, gloeo.I.ci[1,], gloeo.I.ci[3,], col = col.alpha(N.cols[1], trans))
+lines(forecast_time, gloeo.I.ci[2,], lwd=0.5)
+
+###### parameter uncertainty #######
+gloeo.IP <- forecast_gloeo(IC=IC[prow,N],  ## shouldn't I be holding IC constant tho??
```

Comment:
> ## shouldn't I be holding IC constant tho??

You could, but you would have to change your code at the end that calculates the relative variance

Reply:
yes! I realized this after the fact and forgot to take out comment - this makes sense to me now thanks!